sdk/core/src/pipeline.rs

```diff
@@ -24,12 +24,18 @@ use std::sync::Arc;
 /// cannot be enforced by code). All policies except Transport policy can assume there is another following policy (so
 /// self.pipeline[0] is always valid).
 #[derive(Debug, Clone)]
-pub struct Pipeline {
+pub struct Pipeline<R>
```

Comment:
nit: maybe the generic `C` would be a better letter since it's the context.

Also, can we document what the generic is?